pipelines/build/common/openjdk_build_pipeline.groovy

```diff
@@ -999,15 +986,22 @@ class Build {
                 }
             } finally {
                 // post-build workspace clean:
-                //   AIX due to limited ram drive space
-                //   s390x due to limited available disk space on Marist nodes 
-                if (buildConfig.TARGET_OS == "aix" || buildConfig.ARCHITECTURE == "s390x") {
+                if (cleanWorkspaceAfter || cleanWorkspaceBuildOutputAfter) {
                     try {
-                        context.timeout(time: buildTimeouts.AIX_CLEAN_TIMEOUT, unit: "HOURS") {
-                            context.cleanWs notFailBuild: true
+                        context.timeout(time: buildTimeouts.NODE_CLEAN_TIMEOUT, unit: "HOURS") {
+                            // Note: Underlying org.apache DirectoryScanner used by cleanWs has a bug scanning where it misses files containing ".." so use rm -rf instead
+                            if (cleanWorkspaceAfter) {
+                                context.println "Cleaning workspace files: " + context.WORKSPACE + "/*"
+                                context.sh(script: "rm -rf " + context.WORKSPACE + "/*")
+                            } else {
+                                context.println "Cleaning workspace build output files: " + context.WORKSPACE + "/workspace/build/src/build"
+                                context.sh(script: "rm -rf " + context.WORKSPACE + "/workspace/build/src/build")
+                                context.println "Cleaning workspace build output files: " + context.WORKSPACE + "/workspace/target"
+                                context.sh(script: "rm -rf " + context.WORKSPACE + "/workspace/target")
+                            }
                         }
                     } catch (FlowInterruptedException e) {
-                        context.println "[ERROR] AIX clean workspace timeout (${buildTimeouts.AIX_CLEAN_TIMEOUT} HOURS) has been reached. Exiting..."
+                        context.println "[ERROR] clean workspace timeout (${buildTimeouts.NODE_CLEAN_TIMEOUT} HOURS) has been reached. Exiting..."
```

Comment:
Is it always a timeout?

Reply:
I think that was originally added if cleanWs could hang/loop, maybe on Windows? not sure really. Can't see an awful lot of point of it. If the workspace clean fails, we really should fail the job, as the workspace could be in an odd state.
I'll change that.

Reply:
ah just looked again, so yes, FlowInterruptedException is a Interrupt from the context.timeout(), so it is correct. A different failure/Exception is caught by the outer catch(Exception e)...